corehq/pillows/case_search.py

```diff
@@ -98,22 +98,21 @@ def _get_case_properties(doc_dict):
     return base_case_properties + dynamic_mapping
 
 
-class CaseSearchPillowProcessor(ElasticProcessor):
-
-    def process_change(self, change):
-        assert isinstance(change, Change)
-        if self.change_filter_fn and self.change_filter_fn(change):
-            return
+class CaseSearchPillowProcessor(BulkElasticProcessor):
 
+    def _is_case_search_domain(self, change):
         if change.metadata is not None:
             # Comes from KafkaChangeFeed (i.e. running pillowtop)
             domain = change.metadata.domain
         else:
             # comes from ChangeProvider (i.e reindexing)
             domain = change.get_document()['domain']
```

Comment:
does this negate the benefits of the bulk processor? Looks like it will pull every document of a reindex individually. I wonder if the `ChangeProvider` can add the domain in the metadata too, so you still get the bulk transaction benefits?

Reply:
It still groups the writes, which I imagine is a large benefit, but yeah, its definitely not as nice for those. That doesn't seem like a big change to make, and is probably nice for other pillows as well.

Reply:
I did a bit more digging, and it looks like that is only for the couch `ChangeProvider`, reindexing sql cases will properly fill the change metadata, so I think this should be fine to leave as is.

Reply:
Awesome. Thanks for following up with that.